### PR TITLE
Integrate ImperativeOutScale into ImperativeQuantAware.

### PR DESCRIPTION
**Pull request by gfwm2013**
### PR types
Others

### PR changes
Others

### Describe
The main changes of this PR are:

- Add the function to skip specified layers in imperative quantization.
- Integrate ImperativeOutScale into ImperativeQuantAware. 
- Optimize the unittest test_imperative_out_scale.

### CODE REVIEW

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py [hunk 1]
```diff
@@ -59,7 +60,8 @@ def __init__(self,
                  weight_quantize_type='abs_max',
                  activation_quantize_type='moving_average_abs_max',
                  moving_rate=0.9,
-                 quantizable_layer_type=['Conv2D', 'Linear']):
+                 quantizable_layer_type=['Conv2D', 'Linear'],
+                 is_calc_out_scale=False):
```
Comment (juncaipeng) [inline python/paddle/fluid/contrib/slim/quantization/imperative/qat.py]: For deploying the quantized model on CPU and GPU, QAT has to calculate out threshold for all quantized ops, so please delete input params `is_calc_out_scale `.

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py [hunk 2]
```diff
@@ -44,6 +44,7 @@
     "tanh": [["X"], ["Out"]],
     "batch_norm": [["X"], ["Y"]],
     "sigmoid": [["X"], ["Out"]],
+    "swish": [["X"], ["Out"]],
```
Comment (juncaipeng) [inline python/paddle/fluid/contrib/slim/quantization/imperative/qat.py:49]: 这个建议导入quantization/quantization_pass.py中的_op_real_in_out_name

统计输出scale的类中有一个求op输入输出名字的函数，也是可以复用。

**Review by juncaipeng — COMMENTED** [review]
统计输出scale的类，需要考虑不同的bn和不同的pool，建议汇总一下paddle.nn下的op

**Review by juncaipeng — DISMISSED** [review]
LGTM

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py (hunk 2 above)
Comment (gfwm2013) [reply python/paddle/fluid/contrib/slim/quantization/imperative/qat.py:49]: 求op输入输出名字函数已经复用，不过因动静态图所支持的op数量不同，故_op_real_in_out_name在此处未复用。

File: python/paddle/fluid/contrib/slim/quantization/imperative/qat.py (hunk 1 above)
Comment (gfwm2013) [reply python/paddle/fluid/contrib/slim/quantization/imperative/qat.py]: Done.

**Review by juncaipeng — APPROVED** [review]
LGTM

**Pull request merged**